File: 01.4_router/router_b.js
var zmq = require('zmq');

var frontend = zmq.socket('router');

frontend.identity = 'router2';
frontend.linger = 1000;
frontend.reconnect_ivl = 1000;

frontend.bind('tcp://127.0.0.1:6001', function(err) {
  if (err) throw err;

  frontend.on('message', function(conn_id, delimiter, data) {
    //console.log([conn_id, delimiter, data]);
    console.log(data.toString());
    frontend.send([conn_id, delimiter, data + " " + process.pid]);
  });
});

File: 01.4_router/ruby/router_b.rb
require 'ffi-rzmq'

ctx = ZMQ::Context.new

frontend = ctx.socket ZMQ::ROUTER

frontend.bind("tcp://127.0.0.1:8114")

loop do
  msg = []
  frontend.recv_strings(msg)
  puts msg.inspect
  frontend.send_strings([msg[0], msg[1], "Response to: #{msg[2]}"])
end
